docs/architecture/venice_write_path.md

```diff
@@ -0,0 +1,42 @@
+---
+layout: default
+title: Venice Write Path
+parent: Venice Architecture
+permalink: /docs/architecture/venice_write_path
+---
+
+# Introduction
+
+As a high performance derived data storage platform, Venice is designed to ingest large amount of writes from a variety 
+of sources while providing low latency read access to the data. These characteristics are achieved by a combination of
+design choices and optimizations in the write path. This document describes some internal details of the Venice write path. 
+
+Note that this page is still under construction and only provides glimpses of the Venice writing path. More details will 
+be added in the future.
+
+### Terminology
+
+| Name  | Description                                                                                                 |
+|-------|-------------------------------------------------------------------------------------------------------------|
+| VT    | Version Topic. The version topic is a kafka topic where holds all the data for a given store.               |
+| RT    | Real time Topic. A real time topic is also a kafka topic but it only contains data from nearline producers. |
+| ZK    | Apache ZooKeeper. It's used in multiple Venice components to persist Venice metadata.                       |
+| Helix | Apache Helix. It's used in Venice to manage the cluster and assign partitions to different servers.         |
+| EV    | Helix External View. Venice uses it to represent the state of a server in a leader/follower mode.           |
+| CV    | Helix Customized View. Venice uses it to represent the ingestion state of a server for a given store.       |
+
+### Venice Store Ingestion on server side
+
+In Venice, a table or the dataset is called a store and each store has multiple partitions. Helix assign leaders and
+followers status on each partition to different servers. When assignments are done, each server will create multiple
+`StoreIngestionTask`s to handle ingestion for the partitions it's assigned to. The diagram belows describes the data flow of how a 
+store ingestion task's created and ingest data on the server side.
+
+![Server Ingestion Diagram](../assets/images/server_ingestion.svg)
```

Comment:
This diagram is fully accurate AFAICT. Thanks a lot for putting it together!

Seeing it makes me want to refactor things a bit because the SIT is a little bit of a God class at this point, and the code would probably be easier to follow if the data flow did not go right then back to the left again. That being said, it is good to document the current state and that diagram succeeds in doing that.

Reply:
I thought about this a few times but decided to keep this way because I think if contributors need to write codes for SITs - unless we are going to overhaul it soon - they need to understand this state. I personally got confused by this God class for a while and when I drew the diagram, it's way more clear to me.